storage/flux/table.gen.go.tmpl

```diff
@@ -797,6 +778,141 @@ func (t *{{.name}}GroupTable) advance() bool {
 	return true
 }
 
+type {{.name}}AggregateMethod func([]int64, []{{.Type}}) (int64, {{.Type}})
+
+// determine{{.Name}}AggregateMethod returns the method for aggregating
+// returned points within the same group. The incoming points are the
+// ones returned for each series and the method returned here will
+// aggregate the aggregates.
+func determine{{.Name}}AggregateMethod(agg datatypes.Aggregate_AggregateType) ({{.name}}AggregateMethod, error){
+ 	switch agg {
+	case datatypes.AggregateTypeFirst:
+		return groupByFirst{{.Name}}, nil
+	case datatypes.AggregateTypeLast:
+		return groupByLast{{.Name}}, nil
+	case datatypes.AggregateTypeCount:
+		{{if eq .Name "Integer"}}
+		return groupByCount{{.Name}}, nil
+		{{else}}
+		return nil, &influxdb.Error {
+			Code: influxdb.EInvalid,
+			Msg: "unsupported for aggregate count: {{.Name}}",
+		}
+		{{end}}
+	case datatypes.AggregateTypeSum:
+		{{if and (ne .Name "Boolean") (ne .Name "String")}}
+		return groupBySum{{.Name}}, nil
+		{{else}}
+		return nil, &influxdb.Error {
+			Code: influxdb.EInvalid,
+			Msg: "unsupported for aggregate sum: {{.Name}}",
+		}
+		{{end}}
+	case datatypes.AggregateTypeMin:
+		{{if and (ne .Name "Boolean") (ne .Name "String")}}
+		return groupByMin{{.Name}}, nil
+		{{else}}
+		return nil, &influxdb.Error {
+			Code: influxdb.EInvalid,
+			Msg: "unsupported for aggregate min: {{.Name}}",
+		}
+		{{end}}
+	case datatypes.AggregateTypeMax:
+		{{if and (ne .Name "Boolean") (ne .Name "String")}}
+		return groupByMax{{.Name}}, nil
+		{{else}}
+		return nil, &influxdb.Error {
+			Code: influxdb.EInvalid,
+			Msg: "unsupported for aggregate max: {{.Name}}",
+		}
+		{{end}}
+	default:
+		return nil, &influxdb.Error {
+			Code: influxdb.EInvalid,
+			Msg: fmt.Sprintf("unknown/unimplemented aggregate type: %v", agg),
+		}
+	}
+}
+
+{{if and (ne .Name "Boolean") (ne .Name "String")}}
+func groupByMin{{.Name}}(timestamps []int64, values []{{.Type}}) (int64, {{.Type}}) {
+	value := values[0]
+	timestamp := timestamps[0]
+
+	for i := 1; i < len(values); i++ {
+		if value > values[i] {
+			value = values[i]
+			timestamp = timestamps[i]
+		}
+	}
+
+	return timestamp, value
+}
+{{end}}
+
+{{if and (ne .Name "Boolean") (ne .Name "String")}}
+func groupByMax{{.Name}}(timestamps []int64, values []{{.Type}}) (int64, {{.Type}}) {
+	value := values[0]
+	timestamp := timestamps[0]
+
+	for i := 1; i < len(values); i++ {
+		if value < values[i] {
+			value = values[i]
+			timestamp = timestamps[i]
+		}
+	}
+
+	return timestamp, value
+}
+{{end}}
+
+// For group count and sum, the timestamp here is always math.MaxInt64.
+// their final result does not contain _time, so this timestamp value can be anything
+// and it won't matter.
+{{if eq .Name "Integer"}}
+func groupByCount{{.Name}}(timestamps []int64, values []{{.Type}}) (int64, {{.Type}}) {
```

Comment:
Do we need this? Can we just call `groupBySum` directly everywhere this would be called? It might make this a little easier to understand.

Reply:
I don't think it's needed, but I didn't want it to be confusing. Do you think it would be better named as `aggregateCountGroups`? That might represent more of what this is. It is an aggregation of existing groups rather than the grouping themselves.

Reply:
Yes, I like that name just fine.